dxr/plugins/clang/menus.py

```diff
@@ -198,33 +200,57 @@ def _more_menu_items(self, (qualname,)):
                'icon': 'reference'}
 
 
-class FunctionRef(_RefWithDefinition):
+class FunctionRef(_ClangRef):
     """Ref for function definitions or references"""
 
     @classmethod
-    def _condensed_menu_data(cls, tree, prop):
-        return prop['qualname'], 'has_overriddens' in prop, 'has_overrides' in prop
+    def from_condensed(cls, tree, prop):
+        # We ignore the def location clang gives us since it can be wrong, but
+        # its existence means this ref isn't itself a def.
+        search_for_def = prop.get('defloc') is not None
+        return cls(tree,
+                   (search_for_def, prop['qualname'],
+                    'has_overriddens' in prop, 'has_overrides' in prop),
+                   qualname=prop['qualname'])
+
+    def menu_items(self):
+        search_for_def, qualname = self.menu_data[:2]
+        if search_for_def:
+            menu = [{'html': "Jump to definition",
+                     'title': "Jump to definition",
+                     'href': url_for('.search', tree=self.tree.name,
```

Comment:
IIRC url_for is a big time sink to call repeatedly, we may want to hardcode this particular construction like search_url does.